delta/plugins/archive/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/archive/model/ArchiveReference.scala

```diff
@@ -0,0 +1,77 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.archive.model
+
+import cats.Order
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+
+import java.nio.file.Path
+
+/**
+  * Enumeration of archive references.
+  */
+sealed trait ArchiveReference extends Product with Serializable {
+
+  /**
+    * @return the referenced resource id optionally qualified with a tag or a revision
+    */
+  def ref: ResourceRef
+
+  /**
+    * @return the parent project of the referenced resource
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the target location in the archive
+    */
+  def path: Path
+
+  /**
+    * @return the archive reference type
+    */
+  def tpe: ArchiveReferenceType
+}
+
+object ArchiveReference {
+
+  /**
+    * An archive resource reference.
+    *
+    * @param ref            the referenced resource id
+    * @param project        the parent project of the referenced resource
+    * @param path           the target location in the archive
+    * @param representation the format in which the resource should be represented
+    */
+  final case class ResourceReference(
```

Comment:
isn't it missing the field [originalSource](https://bluebrainnexus.io/docs/delta/api/current/kg-archives-api.html#archive-format) ?

Reply:
The resource reference has a representation field that can be: source / expanded / compacted / ntriples or dot.
All are mutually exclusive so this makes for a better model.

Same thing with the ResourceRef field: it can be Latest / Tag / Rev...

Reply:
The wire representation will include that field and possibly a new one like `format`.